### ISSUE TITLE
Malformed `// flow-typed version:` in file header throws error when using `flow-typed install`

### ISSUE THREAD

**Issue by Sparragus** [flowtype/flow-typed#966]
When `flow-typed install`ing `lodash@4.x.x` for `flow-bin@^0.46.0`, the `flow-typed version` at the top of the libdef contains a space (`' '`) that is breaking future `flow-typed install`s.

This error is preventing from using `flow-typed` after the `lodash` libdef has been installed.

I'm using `flow-typed@2.1.2` and `node@8.0.0`.

### How to reproduce:
1. Clone [this repository](https://github.com/Sparragus/flow-typed-lodash)
2. `npm install`
3. Run `npm start` install the libdefs.
4. Run `npm start` again.

On the fourth step you will see the error.

If you `less flow-typed/npm/lodash_v4.x.x.js`, you will be able to see the space for yourself.

The header lines look like this:
```
// flow-typed signature: 53b581d81115b53eaf3a74ae963c4f43
// flow-typed version: ba3c3d2dca/lodash_v4.x.x/flow_>=v0.38.x <=v0.46.x
                                                              ↳  Here is the space!

declare module 'lodash' {
  ...
}
```

The error is:
```
$ flow-typed install --overwrite
• Found 2 dependencies in package.json to install libdefs for. Searching...
• rebasing flow-typed cache...
UNCAUGHT ERROR: Error: flow_v0.38.x <=v0.46.x: Unexpected trailing characters:  <=v0.46.x
    at validationError (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/dist/lib/validationErrors.js:14:11)
    at parseDirString (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/dist/lib/flowVersion.js:155:45)
    at _callee2$ (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/dist/lib/npm/npmLibDefs.js:502:146)
    at tryCatch (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/node_modules/regenerator-runtime/runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as next] (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/node_modules/regenerator-runtime/runtime.js:117:21)
    at tryCatch (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/node_modules/regenerator-runtime/runtime.js:65:40)
    at invoke (/Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/node_modules/regenerator-runtime/runtime.js:155:20)
    at /Users/richard/.nvm/versions/node/v8.0.0/lib/node_modules/flow-typed/node_modules/regenerator-runtime/runtime.js:165:13
    at <anonymous>
```

**Comment by leoselig** [comment]
Probably the same as https://github.com/flowtype/flow-typed/issues/913
(not resolved)

**Comment by renatorib** [comment]
```
> node --version
v7.7.4
```

```
> flow-typed version
flow-typed@2.1.2
```
```
> flow-typed install
• Found 56 dependencies in package.json to install libdefs for. Searching...
• rebasing flow-typed cache...
UNCAUGHT ERROR: Error: flow_v0.38.x <=v0.46.x: Unexpected trailing characters:  <=v0.46.x
```

**Comment by joeybaker** [comment]
FWIW, I fixed a similar issue with `rm -rf flow-typed/npm && flow-typed install`

**Comment by dreid** [comment]
This should be fixed in v2.2 by #1311
